server/main.lua

```diff
@@ -1,62 +1,60 @@
 local sharedConfig = require 'config.shared'
 
-lib.callback.register('qb-weed:server:getBuildingPlants', function(_, building)
+lib.callback.register('qbx_weed:server:getBuildingPlants', function(_, building)
     local buildingPlants = {}
     local plants = MySQL.query.await('SELECT * FROM house_plants WHERE building = ?', { building })
 
     for i = 1, #plants, 1 do
         local plant = plants[i]
-        if type(plant.coords) == 'string' then
-            plant.coords = json.decode(plant.coords)
-            plant.coords = vec3(plant.coords.x, plant.coords.y, plant.coords.z)
-        end
+        plant.coords = json.decode(plant.coords)
+        plant.coords = vec3(plant.coords.x, plant.coords.y, plant.coords.z)
         buildingPlants[#buildingPlants + 1] = plant
     end
 
     return buildingPlants
 end)
 
-RegisterNetEvent('qb-weed:server:placePlant', function(coords, sort, currentHouse)
-    local random = math.random(1, 2)
-    local gender = 'woman'
-    if random == 1 then
-        gender = 'man'
-    end
-    MySQL.insert.await('INSERT INTO house_plants (building, coords, gender, sort, plantid) VALUES (?, ?, ?, ?, ?)', { currentHouse, json.encode(coords), gender, sort, math.random(111111, 999999) })
-    TriggerClientEvent('qb-weed:client:refreshHousePlants', -1, currentHouse)
+RegisterNetEvent('qbx_weed:server:placePlant', function(coords, sort, currentHouse)
+    local gender = math.random(1, 2) == 1 and 'female' or 'male'
+    MySQL.insert.await('INSERT INTO house_plants (building, coords, gender, sort) VALUES (?, ?, ?, ?)', { currentHouse, json.encode(coords), gender, sort })
+    TriggerClientEvent('qbx_weed:client:refreshHousePlants', -1, currentHouse)
 end)
 
-RegisterNetEvent('qb-weed:server:removeDeathPlant', function(building, plantId)
-    MySQL.query.await('DELETE FROM house_plants WHERE plantid = ? AND building = ?', { plantId, building })
-    TriggerClientEvent('qb-weed:client:refreshHousePlants', -1, building)
+RegisterNetEvent('qbx_weed:server:removeDeathPlant', function(building, plantId, plantCoords)
+    local src = source
+    local player = exports.qbx_core:GetPlayer(src)
+    if not player or not building then return end
+    if #(GetEntityCoords(GetPlayerPed(src)) - plantCoords) > 2 then return end
+    MySQL.prepare.await('DELETE FROM house_plants WHERE id = ?', { plantId })
+    TriggerClientEvent('qbx_weed:client:refreshHousePlants', -1, building)
 end)
```

Comment:
```suggestion
    local player = exports.qbx_core:GetPlayer(source)
    if not player or not building then return end
    if #(GetEntityCoords(GetPlayerPed(player.PlayerData.source)) - plantCoords) > 2 then return end
    MySQL.prepare.await('DELETE FROM house_plants WHERE id = ?', { plantId })
    TriggerClientEvent('qbx_weed:client:refreshHousePlants', -1, building)
end)
```
Here we want to get rid of these local source variables, they are unneeded as soon as we grab the Player object as its already cached there.